src/components/tl/ucp/allgather/allgather_ring.c

```diff
@@ -43,15 +43,18 @@ void ucc_tl_ucp_allgather_ring_progress(ucc_coll_task_t *coll_task)
     ucc_rank_t         sendto, recvfrom, sblock, rblock;
     int                step;
     void              *buf;
+    int                use_loopback = UCC_TL_UCP_TEAM_LIB(team)->cfg.allgather_use_loopback;
 
     if (UCC_INPROGRESS == ucc_tl_ucp_test(task)) {
         return;
     }
     sendto   = ucc_ep_map_eval(task->subset.map, (trank + 1) % tsize);
     recvfrom = ucc_ep_map_eval(task->subset.map, (trank - 1 + tsize) % tsize);
 
-    while (task->tagged.send_posted < tsize - 1) {
-        step = task->tagged.send_posted;
+    int iter = use_loopback ? tsize : tsize - 1; //when using loopback tagged.send_posted has 1 more which will cause non-complete ring algorithm
```

Comment:
declaration of variable goes to the beginning of the block

Reply:
I am not sure to understand the comment. What do you mean by "will cause non-complete ring algorithm"? 
Can you use a more explicit name than `iter` for that variable?